assignment/tool/src/java/org/sakaiproject/assignment/tool/AssignmentAction.java

```diff
@@ -3581,6 +3604,19 @@ protected String build_instructor_grade_submission_context(VelocityPortlet portl
                     }
                     context.put("value_grades", grades);
                 }
+
+                // Check if the assignment has a rubric associated or not
+                context.put(RubricsConstants.RBCS_HAS_ASSOCIATED_RUBRIC, rubricsService.hasAssociatedRubric(RubricsConstants.RBCS_TOOL_ASSIGNMENT, a.getId()));
+                if (assignmentToolUtils.hasRubricSelfReview(context, a.getId())) {
+                	s.getSubmitters().stream().findAny().ifPresent(u -> context.put(STUDENT_WITH_SELFREPORT, u.getSubmitter()));
+                }
+                // Only one peer review for an assignment submission per student.
+                if (a.getAllowPeerAssessment()) {
+                    List<PeerAssessmentItem> reviews = assignmentPeerAssessmentService.getPeerAssessmentItems(s.getId(), a.getScaleFactor());
+                    if(reviews.size() > 1) { log.warn("More than one peer review were found for Rubric assignment"); }
```

Comment:
There could be reviews for multiple submissions, it's possible. If not now, in future.

Reply:
The limitation to 1 peer-review per assignment is for two reasons: our client asked us that way and the second reason so that the teacher's screen can be clearly qualified and because in the repository only 1 evaluator allows (already either by submission or by student id).

If in the future someone wants something else, it is already outside the scope of this ticket, it would be a change like any other.